apps/daimo-mobile/src/view/screen/receive/SendRequestScreen.tsx

```diff
@@ -31,11 +31,15 @@ export default function SendRequestScreen() {
   );
   const nav = useNav();
   const trackRequest = (requestId: `${bigint}`) => {
-    account.trackedRequests.push({
-      requestId,
-      amount: `${dollarsToAmount(dollars)}`,
-    });
-    setAccount(account);
+    const newAccount = { ...account };
+    newAccount.trackedRequests = [
```

Comment:
you could also use `newAccount.trackedRequest.push(...` here & similar above but fine either way

Reply:
I think that would violate the rule of hooks: `newAccount.trackedRequest` is a reference shared with `account.trackedRequest` since we shallow copy it above. `.push` would edit the previous state, which is bad in react?